### PR TITLE
T18: Add a linter (2nd attempt)

### PR DESCRIPTION
**Pull request by midwestbread**
this is up and running again, I didn't manually lint this time because I'm 4 commits behind main so I think manually linting would just cause a bunch of merge conflicts, hopefully we can merge and then do it this time.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        // "rootDirs": [],                                   /* Allow multiple folders to be treated as one when resolving modules. */
+        // "typeRoots": [],                                  /* Specify multiple folders that act like './node_modules/@types'. */
+        // "types": [],                                      /* Specify type package names to be included without being referenced in a source file. */
+        // "allowUmdGlobalAccess": true,                     /* Allow accessing UMD globals from modules. */
+        // "moduleSuffixes": [],                             /* List of file name suffixes to search when resolving a module. */
+        // "resolveJsonModule": true,                        /* Enable importing .json files. */
+        // "noResolve": true,                                /* Disallow 'import's, 'require's or '<reference>'s from expanding the number of files TypeScript should add to a project. */
+        /* JavaScript Support */
+        "allowJs": true, /* Allow JavaScript files to be a part of your program. Use the 'checkJS' option to get errors from these files. */
```
Comment (justinpchen94) [inline tsconfig.json]: nit: let's set this to false! we should only be using ts files ideally

**Review by justinpchen94 — COMMENTED** [review]
i think this looks good! but gonna have @chinkevin25 take a pass too

**Comment by justinpchen94** [comment]
Let's also update the title to use the task title so the format of T#: Task Title

File: package.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
+  "devDependencies": {
+    "@typescript-eslint/eslint-plugin": "^5.53.0",
+    "eslint": "^8.34.0",
+    "eslint-config-standard-with-typescript": "^34.0.0",
+    "eslint-plugin-import": "^2.27.5",
+    "eslint-plugin-n": "^15.6.1",
+    "eslint-plugin-promise": "^6.1.1",
+    "eslint-plugin-react": "^7.32.2",
```
Comment (chinkevin25) [inline package.json:9]: I'm not 100% sure if these are being used? I would imagine they have to be imported / configured in your `.eslintrc.js` but it's possible that you're using  something that already includes these and these are just peer dependencies.

File: package.json [hunk 3]
```diff
@@ -0,0 +1,12 @@
+{
+  "devDependencies": {
+    "@typescript-eslint/eslint-plugin": "^5.53.0",
```
Comment (chinkevin25) [inline package.json:3]: do we need to specify the usage of this anywhere? The docs seem to mention something about setting this as the parser? do you know if that is required?

**Review by chinkevin25 — COMMENTED** [review]
Looks great just want to make sure we're using these rules!

**Review by chinkevin25 — APPROVED** [review]
Approving since it works as is

**Pull request merged**